zebra-chain/src/orchard/keys.rs

```diff
@@ -642,7 +647,46 @@ impl From<IncomingViewingKey> for [u8; 64] {
     }
 }
 
-impl From<FullViewingKey> for IncomingViewingKey {
+impl TryFrom<[u8; 64]> for IncomingViewingKey {
+    type Error = &'static str;
+
+    /// Convert an array of bytes into a [`IncomingViewingKey`].
+    ///
+    /// Returns an error if the encoding is malformed or if it [encodes the scalar additive
+    /// identity, 0][1].
+    ///
+    /// > ivk MUST be in the range {1 .. 𝑞P - 1}
+    ///
+    /// [1]: https://zips.z.cash/protocol/protocol.pdf#orchardinviewingkeyencoding
+    fn try_from(bytes: [u8; 64]) -> Result<Self, Self::Error> {
+        let mut dk_bytes = [0u8; 32];
+        dk_bytes.copy_from_slice(&bytes[..32]);
+        let dk = DiversifierKey::from(dk_bytes);
+
+        let mut ivk_bytes = [0u8; 32];
+        ivk_bytes[..].copy_from_slice(&bytes[32..]);
```

Comment:
Tiny nit for consistency:

```suggestion
        ivk_bytes.copy_from_slice(&bytes[32..]);
```